pandas/core/groupby.py

```diff
@@ -457,8 +457,21 @@ def _selected_obj(self):
         else:
             return self.obj[self._selection]
 
-    def _set_selection_from_grouper(self):
-        """ we may need create a selection if we have non-level groupers """
+    def _clear_group_selection(self):
```

Comment:
call this ``_reset_group_selection``. sorry to be a pain about the names, but naming is very important for consistency.